### PR TITLE
Fix test by preventing chaos during verification

### LINKED ISSUE #4214: Failed tests (12468): TestChaos
The following test appears to have failed:

[#12468](https://circleci.com/gh/cockroachdb/cockroach/12468):

```
E0208 02:03:54.259263 27426 acceptance/cluster/localcluster.go:410  node=0 status=restart
I0208 02:03:54.262949 27426 acceptance/chaos_test.go:162  client 0 shutting down
I0208 02:03:54.426406 27426 acceptance/chaos_test.go:162  client 1 shutting down
I0208 02:03:54.448069 27426 acceptance/chaos_test.go:162  client 2 shutting down
I0208 02:03:54.550401 27426 acceptance/cluster/localcluster.go:538  stopping
--- FAIL: TestChaos (18.39s)
	chaos_test.go:280: dial tcp 0.0.0.0:32787: getsockopt: connection refused
=== RUN   TestGossipPeerings
I0208 02:03:57.242309 27426 acceptance/cluster/localcluster.go:219  started docker-spy: 172.17.0.16
I0208 02:03:57.242338 27426 acceptance/cluster/localcluster.go:226  initializing cluster
I0208 02:03:59.100442 27426 acceptance/cluster/localcluster.go:333  creating ca (1024bit) in: /home/ubuntu/cockroach/.localcluster.certs.822025371
I0208 02:03:59.158545 27426 acceptance/cluster/localcluster.go:338  creating node (1024bit) certs in: /home/ubuntu/cockroach/.localcluster.certs.822025371
I0208 02:04:01.466503 27426 acceptance/cluster/localcluster.go:389  *** started roach0.local ***
  ui:    https://0.0.0.0:32789
  trace: https://0.0.0.0:32789/debug/requests
  logs:  /tmp/circle-artifacts.c2nnytD/acceptance/TestGossipPeerings/roach0.local/cockroach.INFO
--
--- SKIP: TestBuildBabyCluster (0.00s)
	terraform_test.go:32: only enabled during testing
=== RUN   TestFiveNodesAndWriters
--- SKIP: TestFiveNodesAndWriters (0.00s)
	main_test.go:62: running in docker mode
FAIL
ok github.com/cockroachdb/cockroach/acceptance 1337s

```
Please assign, take a look and update the issue accordingly.

### PR DESCRIPTION
**Pull request by vivekmenezes**
closes #4214 

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/4225)

### CODE REVIEW

File: acceptance/chaos_test.go [hunk 1]
```diff
@@ -272,13 +272,14 @@ CREATE TABLE bank.accounts (
 
 	// Verify accounts.
 
-	// Recreate db client because node 0 might have been restarted by
+	// Hold the read lock on node 0 to prevent it being restarted by
 	// chaos monkey.
-	db = makePGClient(t, c.PGUrl(0))
+	clients[0].RLock()
```
Comment (tamird) [inline acceptance/chaos_test.go]: does the monkey hold the lock until the node comes back?

can you put this below `var sum int`?

**Comment by tamird** [comment]
LGTM

**Comment by tamird** [comment]
Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4225)*

**Comment by BramGruneir** [comment]
LGTM

**Comment by vivekmenezes** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[acceptance/chaos_test.go, line 277 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4225#-KA0ZBaUTIksALYsdKxs-r1-277:-KA1JkHYYJKzwEL1BIB-:-1163438884) ([raw file](https://github.com/cockroachdb/cockroach/blob/af3e6e3a27b7119144283d076b38ea400cb1f7ca/acceptance/chaos_test.go#L277)):*
The monkey restarts a few randomly selected nodes, followed by recreating the connections to these new nodes. I don't think there are any guarantees here other than the fact that when using a connection while holding a read lock there is the guarantee that the monkey cannot restart the node and so the connection is a valid one to use, and thus the request being sent on the connection will ultimately succeed.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4225)*

**Comment by bdarnell** [comment]
Could we solve this instead with a retry loop? Clients will in general need to be able to handle nodes that die, so it seems better to handle this case than to prevent it from happening (unless that's too hard to do right now with the `lib/pq` driver; we might have to devise some way to get it to select from multiple nodes)

---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4225#-:-KA1WMZUBOV07ZGK4saf:351162766)*

**Comment by vivekmenezes** [comment]
Yeah, I did see it fail again after this fix :-(

**Pull request merged**